client/extensions/woocommerce/app/order/order-details/table.js

```diff
@@ -133,19 +133,36 @@ class OrderDetailsTable extends Component {
 		this.props.onChange( { shipping_lines: [ { ...shippingLine, total } ] } );
 	};
 
-	onDelete = ( id, type = 'line_items' ) => {
-		return () => {
-			const index = findIndex( this.props.order[ type ], { id } );
-			if ( index >= 0 ) {
-				let newItem;
-				if ( 'line_items' === type ) {
-					newItem = { id, quantity: 0, subtotal: 0 };
-				} else {
-					newItem = { id, name: null, total: 0 };
-				}
-				this.props.onChange( { [ type ]: { [ index ]: newItem } } );
+	onDelete = ( id, type = 'line_items' ) => () => {
+		const index = findIndex( this.props.order[ type ], { id } );
+		if ( index >= 0 ) {
+			let newItem;
+			if ( 'line_items' === type ) {
+				newItem = { id, quantity: 0, subtotal: 0 };
+			} else {
+				newItem = { id, name: null, total: 0 };
 			}
-		};
+			this.props.onChange( { [ type ]: { [ index ]: newItem } } );
+		}
+	};
+
+	removeCoupon = item => () => {
```

Comment:
So we are leaving this here for now even though it isn't being used until #21280 lands

Reply:
No, I just missed it 🤦‍♀️ – removed in 45b06df80ce2895b07f123ecad958e71f14ff257